cursor.go

```diff
@@ -151,6 +151,20 @@ func (c *Cursor) Delete() error {
 	}
 	c.node().del(key)
 
+	pos := c.curPos()
+	// If the deleted item is the first item in the node, we should keep
+	// the cursor's position (decrease by 1). Otherwise, the cursor will
+	// automatically move to the next item, and when clients call `Next`
+	// method afterward, it will skip one item in the node.
+	//
+	// Note if there isn't any write on current page in current transaction,
+	// then the cursor only iterates items against the page instead of the
+	// in-memory node. Since the page is immutable, so we don't need to change
+	// pos.index in such case.
+	if pos != nil && pos.index == 0 && pos.node != nil {
+		pos.index--
```

Comment:
nit: that's personal preferences but I would just set this to -1 directly. Takes a little mental math off my head when reading.

Reply:
updated.